omniledger/service/service_test.go

```diff
@@ -198,14 +199,90 @@ func TestService_GetProof(t *testing.T) {
 	require.NotNil(t, err)
 }
 
+func TestService_WaitInclusion(t *testing.T) {
+	for i := 0; i < 3; i++ {
+		log.Lvl1("Testing inclusion when sending to service", i)
+		waitInclusion(t, i)
+	}
+}
+
+func waitInclusion(t *testing.T, client int) {
+	s := newSer(t, 2, testInterval)
+	defer s.local.CloseAll()
+	defer closeQueues(s.local)
+
+	// Create a transaction without waiting
+	log.Lvl1("Create transaction and don't wait")
+	pr := sendTransaction(t, s, client, dummyKind, 0)
+	require.False(t, pr.InclusionProof.Match())
+
+	// Create a transaction and wait
+	log.Lvl1("Create correct transaction and wait")
+	pr = sendTransaction(t, s, client, dummyKind, 10)
+	require.True(t, pr.InclusionProof.Match())
+
+	// Create a failing transaction and wait
+	done := make(chan bool)
+	go func() {
+		log.Lvl1("Create wrong transaction and wait")
+		pr := sendTransaction(t, s, client, invalidKind, 10)
+		require.False(t, pr.InclusionProof.Match())
+		done <- true
+	}()
+
+	// Wait two intervals before sending a new transaction
+	time.Sleep(2 * s.interval)
+
+	// Create a transaction and wait
+	log.Lvl1("Create second correct transaction and wait")
+	pr = sendTransaction(t, s, client, dummyKind, 5)
+	require.True(t, pr.InclusionProof.Match())
+	select {
+	case <-done:
+		require.Fail(t, "go-routine should not be done yet")
+	default:
+	}
+
+	<-done
+}
+
+func sendTransaction(t *testing.T, s *ser, client int, kind string, wait int) Proof {
+	tx, err := createOneClientTx(s.darc.GetBaseID(), kind, s.value, s.signer)
+	require.Nil(t, err)
+	var w *int
+	if wait > 0 {
+		w = &wait
+	}
+	ser := s.services[client]
+	_, err = ser.AddTransaction(&AddTxRequest{
+		Version:       CurrentVersion,
+		SkipchainID:   s.sb.SkipChainID(),
+		Transaction:   tx,
+		InclusionWait: w,
+	})
```

Comment:
this part can be replaced by `ser.SendTxTo` so that we don't repeat ourselves. But you'll have to change `ser.SendTxTo` to accept `InclusionWait`.